exp/controllers/azureasomanagedcontrolplane_controller_test.go

```diff
@@ -224,7 +224,8 @@ func TestAzureASOManagedControlPlaneReconcile(t *testing.T) {
 				Namespace: cluster.Namespace,
 			},
 			Status: asocontainerservicev1.ManagedCluster_STATUS{
-				Fqdn: ptr.To("endpoint"),
+				Fqdn:                     ptr.To("endpoint"),
+				CurrentKubernetesVersion: ptr.To("Current"),
```

Comment:
nit: Changing "Current" to a string in semver format might make more sense so we don't have to check for "vCurrent".

Reply:
What exactly do you mean by "so we don't have to check for 'vCurrent'"?

Reply:
Looks like there is some "v" prefix injection here. If so should we have two tests, one where we validate the "v" injection (passing in a real version in the test case e.g., "1.30.0" makes sense IMO) and then one where we pass in a string that starts w/ "v" (e.g., "v1.30.0") and verify that there isn't double-injection (e.g., "vv1.30.0")

Reply:
AFAIK AKS never adds the "v" prefix, but should we account for that anyway?

Reply:
Ah I was just being extra picky about the formatting of the string and passing in a real version. I don't think there's anything wrong with keeping it as "Current".

Also if AKS never adds it, I think it should be good as is.